tests/std/tests/P0466R5_layout_compatibility_and_pointer_interconvertibility_traits/test.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#include <assert.h>
+#include <iostream>
+#include <type_traits>
+
+using namespace std;
+
+#define ASSERT(...) assert((__VA_ARGS__))
+
+constexpr bool test() {
+
+    // is_layout_compatible tests
+    {
+        struct S {
+            int v1;
+            int v2;
+        };
+
+        struct S1 {
+            S s1;
+            int v3;
+        };
+
+        struct S2 {
+            S s1;
+            int v2;
+        };
+
+        struct S3 {
+            S s1;
+            int v2;
+            int v3;
+        };
+
+        enum E1 { e1, e2, e3 };
+
+        enum E2 : int { e4, e5, e6 };
+
+        enum E3 : unsigned int { e7, e8, e9 };
+
+        ASSERT(is_layout_compatible_v<int, int>);
+        ASSERT(is_layout_compatible_v<E1, E2>);
+        ASSERT(is_layout_compatible_v<S1, S2>);
+
+        ASSERT(!is_layout_compatible_v<int, char>);
+        ASSERT(!is_layout_compatible_v<E1, E3>);
+        ASSERT(!is_layout_compatible_v<S1, S3>);
+    }
+
+    // is_pointer_interconvertible_base_of tests
+    {
+        class A {};
+        class B : public A {};
+        class C : public A {
+            int : 0;
+        };
+// Disable warning C4408: anonymous union did not declare any data members
+#pragma warning(push)
+#pragma warning(disable : 4408)
+        class D : public A {
+            union {};
+        };
+#pragma warning(pop)
+        class E : public A {
+            int v1 = 2;
+        };
+
+        class A1 : public A {};
+        class A2 : public A {};
+        class A3 : public A2 {};
+        class A4 : A1, A3 {};

```

Comment:
Why'd we switch from public to private inheritance here? It shouldn't make any difference since the trait sees private bases (as does `is_base_of`), so the anomaly is suspect.

Reply:
Fixed, made consistent